agent/doc/swdesign/README.md

```diff
@@ -1315,6 +1336,86 @@ Needs:
 - impl
 - utest
 
+#### WorkloadControlLoop handles restarts of workloads
+`swdd~workload-control-loop-handles-workload-restarts~1`
+
+Status: approved
+
+When the WorkloadControlLoop receives a new workload state and the WorkloadControlLoop checks if the restart of the workload it manages is required, then the WorkloadControlLoop shall execute the restart of the workload it manages based on the result of the comparison
+
+Rationale:
+The execution of a restart of the workload depends on the workload state and the configured restart policy.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+
+#### WorkloadControlLoop skips restarts
+`swdd~workload-control-loop-skips-restarts~1`
+
+When the WorkloadControlLoop handles a restart of a workload and the workload's current `WorkloadInstanceName` inside the WorkloadControlLoop differs from the `WorkloadInstanceName` inside the received workload state, then the WorkloadControlLoop shall skip the restart.
+
+If the workload changed in the meantime, there is no need to restart it.
+
+Rationale:
+This prevents wrong restarts of workloads when a new workload state of the old workload is received during an update operation runs.
+
+Tags:
+- WorkloadControlLoop
+
+Needs:
+- impl
+- utest
+
+#### WorkloadControlLoop restarts workload with enabled restart policy
+`swdd~workload-control-loop-restarts-workload-with-enabled-restart-policy~1`
+
+Status: approved
+
+When the workload has not changed and:
+* either the workload's `ExecutionState` is `Succeeded(Ok)` or `ExecutionState` is `Failed(ExecFailed)` and the workload's configured `RestartPolicy` contains the value `ALWAYS`
+* or the workload's `ExecutionState` is `Failed(ExecFailed)` and the workload's configured `RestartPolicy` contains the value `ON_FAILURE`,
+then the WorkloadControlLoop shall restart the workload.
+
+Comment:
+If the workload changed in the meantime, there is no need to restart it. In case of the restart policy is `NEVER` the workload is not restarted.
```

Comment:
The first sentence is clarifying a different requirement, we can remove it.
```suggestion
In all other execution states and in case of the restart policy is `NEVER` the workload is not restarted.
```